src/test/java/com/rocketden/main/service/GameManagementServiceTests.java

```diff
@@ -411,6 +411,95 @@ public void submitSolutionEmptyField() {
     }
 
     @Test
+    public void playAgainSuccess() throws Exception {
+        User host = new User();
+        host.setNickname(NICKNAME);
+        host.setUserId(USER_ID);
+        host.setSessionId(SESSION_ID);
+
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+        room.setHost(host);
+        room.setDifficulty(ProblemDifficulty.HARD);
+        room.setActive(true);
+        room.addUser(host);
+        room.setDuration(1L);
+
+        StartGameRequest request = new StartGameRequest();
+        request.setInitiator(UserMapper.toDto(host));
+
+        Mockito.doReturn(room).when(repository).findRoomByRoomId(ROOM_ID);
+        gameService.startGame(ROOM_ID, request);
+
+        // Wait 1 second until the game timeUp socket update is sent
+        Mockito.verify(socketService, Mockito.timeout(1500)).sendSocketUpdate(Mockito.any(GameDto.class));
+
+        PlayAgainRequest playAgainRequest = new PlayAgainRequest();
+        playAgainRequest.setInitiator(UserMapper.toDto(host));
+        RoomDto response = gameService.playAgain(ROOM_ID, playAgainRequest);
+
+        Game game = gameService.getGameFromRoomId(room.getRoomId());
+
+        verify(socketService).sendSocketUpdate(Mockito.eq(GameMapper.toDto(game)));
+
+        assertTrue(game.getPlayAgain());
+        assertEquals(room.getRoomId(), response.getRoomId());
+        assertEquals(room.getDifficulty(), response.getDifficulty());
+        assertFalse(room.getActive());
+        assertNull(room.getHost().getSessionId());
+    }
+
+    @Test
+    public void playAgainWrongInitiator() throws Exception {
+        User host = new User();
+        host.setNickname(NICKNAME);
+        host.setUserId(USER_ID);
+
+        Room room = new Room();
+        room.setRoomId(ROOM_ID);
+        room.setHost(host);
+        room.setDuration(1L);
+
+        StartGameRequest startRequest = new StartGameRequest();
+        startRequest.setInitiator(UserMapper.toDto(host));
+
+        Mockito.doReturn(room).when(repository).findRoomByRoomId(ROOM_ID);
+        gameService.startGame(ROOM_ID, startRequest);
+
+        Mockito.verify(socketService, Mockito.timeout(1500)).sendSocketUpdate(Mockito.any(GameDto.class));
+
+        UserDto initiator = new UserDto();
+        initiator.setNickname(NICKNAME_2);
+        PlayAgainRequest request = new PlayAgainRequest();
+        request.setInitiator(initiator);
+
+        ApiException exception = assertThrows(ApiException.class, () -> gameService.playAgain(ROOM_ID, request));
+        assertEquals(GameError.INVALID_PERMISSIONS, exception.getError());
+    }
+
+    @Test
+    public void playAgainGameNotOver() {
```

Comment:
The first call, `getGameFromRoomId`, should trigger a game not found error (since the game hasn't started yet) - so yeah nothing bad should happen I wouldn't think